tests/views_integration/Tribe/Events/Views/V2/Views/List_ViewTest.php

```diff
@@ -101,6 +102,168 @@ public function test_render_with_upcoming_events() {
 		$this->assertMatchesSnapshot( $html );
 	}
 
+	public function create_event_cat( $tag ) {
+
+		//create categories
+		$cat1 = wp_insert_term( $tag, TEC::TAXONOMY );
+		$term_id = $cat1['term_id'];
+
+		// Sanity check.
+		$term_info = term_exists( $term_id, TEC::TAXONOMY );
+		$this->assertIsArray( $term_info );
+		$this->assertNotEmpty( $term_info );
+
+		return $term_id;
+	}
+
+	/**
+	 * Test render with a category filter
+	 */
+	public function test_render_with_one_category() {
+		$events = [];
+		$term_id = $this->create_event_cat( 'pepperoni' );
+
+		// Create the events.
+		foreach (
+			[
+				'2019-01-01 9am',
+				'2019-01-01 +1 week',
+				'2019-01-01 +9 days',
+			] as $start_date
+		) {
+			$events[] = tribe_events()->set_args( [
+				'start_date' => $start_date,
+				'timezone'   => 'Europe/Paris',
+				'duration'   => 3 * HOUR_IN_SECONDS,
+				'title'      => 'Test Event - ' . $start_date,
+				'status'     => 'publish',
+				'tax_input'  => [
+					'taxonomy' => TEC::TAXONOMY,
+					'field'    => 'term_id',
+					'terms'    => $term_id,
+				],
+			] )->create();
+		}
+
+		foreach( $events as $event ) {
+			// Let's make sure we have the term ID we expect. The above set_args() doesn't always do the trick?
+			wp_set_object_terms( $event->ID, $term_id, TEC::TAXONOMY, true );
+		}
+
+		// Sanity check
+		$this->assertEquals( 3, tribe_events()->where( 'ends_after', $this->mock_date_value )->count() );
+
+		$expected_post_ids = wp_list_pluck( $events, 'ID' );
+
+		foreach ( $expected_post_ids as $id ) {
+			$this->assertTrue( has_term( $term_id, TEC::TAXONOMY, $id ), "Missing term before context change." );
+		}
+
+		$list_view = View::make( List_View::class );
+		$context = tribe_context()->alter( [
+			'event_display'      => 'list',
+			'event_display_mode' => 'list',
+			'today'              => $this->mock_date_value,
+			'now'                => $this->mock_date_value,
+			'event_date'         => $this->mock_date_value,
+			'events_per_page'    => 3,
+			'tax_input'  => [
```

Comment:
This is the part that puzzles me a bit and where I think this fix is not the best one.
In our Repository code we should support the `category` argument for the `where` or `by` method.
Example:

```php
$events = tribe_events()->where('category', 'foo')->get_ids();
```

And, if any fix is required to make that work, then it should be done in the Events repository to make sure it works.

Since the Context and the Repository are "neutral" to unknown inputs, as is the `tax_input` here, they will just pass them through to the query and, sure, the `tax_input` query argument will work.
This seems like a workaround for not working code (the Repository one handling the `category` criteria), and not a fix.

Furthermore, what if the `tax_input` already contains other criteria? Would they be kept or overridden?